packages/autocomplete-js/src/__tests__/detachedMediaQuery.test.ts

```diff
@@ -1,3 +1,46 @@
+import { createMatchMedia, createSource } from '../../../../test/utils';
+import { autocomplete } from '../autocomplete';
+
 describe('detachedMediaQuery', () => {
-  test.todo('tests');
+  afterAll(() => {
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: createMatchMedia({}),
+    });
+  });
+
+  test('falls back to the deprecated `addListener` if `addEventListener` is undefined', () => {
+    const addListener = jest.fn();
+
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: createMatchMedia({
+        matches: true,
+        addListener,
+        addEventListener: undefined,
+      }),
+    });
+
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+    autocomplete<{ label: string }>({
+      id: 'autocomplete',
+      detachedMediaQuery: '',
+      container,
+      getSources() {
+        return [
+          {
+            ...createSource({}),
```

Comment:
You can pass `templates` to the arguments of `createSource`.

Reply:
It's based on the `core` types so we can't pass templates to it

Reply:
Ah right!